presto-main/src/main/java/io/prestosql/sql/planner/SubqueryPlanner.java

```diff
@@ -108,482 +111,318 @@ public PlanBuilder handleSubqueries(PlanBuilder builder, Collection<Expression>
 
     public PlanBuilder handleSubqueries(PlanBuilder builder, Expression expression, Node node)
     {
-        builder = appendInPredicateApplyNodes(builder, collectInPredicateSubqueries(expression, node), node);
-        builder = appendScalarSubqueryCorrelatedJoins(builder, collectScalarSubqueries(expression, node));
-        builder = appendExistsSubqueryApplyNodes(builder, collectExistsSubqueries(expression, node));
-        builder = appendQuantifiedComparisonApplyNodes(builder, collectQuantifiedComparisonSubqueries(expression, node), node);
+        for (List<InPredicate> cluster : cluster(selectSubqueries(builder, expression, analysis.getInPredicateSubqueries(node)))) {
+            builder = planInPredicate(builder, cluster, node);
+        }
+        for (List<SubqueryExpression> cluster : cluster(selectSubqueries(builder, expression, analysis.getScalarSubqueries(node)))) {
+            builder = planScalarSubquery(builder, cluster);
+        }
+        for (List<ExistsPredicate> cluster : cluster(selectSubqueries(builder, expression, analysis.getExistsSubqueries(node)))) {
+            builder = planExists(builder, cluster);
+        }
+        for (List<QuantifiedComparisonExpression> cluster : cluster(selectSubqueries(builder, expression, analysis.getQuantifiedComparisonSubqueries(node)))) {
+            builder = planQuantifiedComparison(builder, cluster, node);
+        }
```

Comment:
There are tests in `TestLogicalPlanner` for duplicate scalar subqueries, duplicate in-predicates and duplicate quantified comparisons. Do we have a test for duplicate EXISTS? Do we have a test for nested subqueries?